cmd/bbolt/main_test.go

```diff
@@ -135,35 +136,76 @@ func TestPageCommand_Run(t *testing.T) {
 }
 
 func TestPageItemCommand_Run(t *testing.T) {
-	db := btesting.MustCreateDBWithOption(t, &bolt.Options{PageSize: 4096})
-	srcPath := db.Path()
-
-	// Insert some sample data
-	t.Log("Insert some sample data")
-	err := db.Fill([]byte("data"), 1, 100,
-		func(tx int, k int) []byte { return []byte(fmt.Sprintf("key_%d", k)) },
-		func(tx int, k int) []byte { return []byte(fmt.Sprintf("value_%d", k)) },
-	)
-	require.NoError(t, err)
-
-	defer requireDBNoChange(t, dbData(t, srcPath), srcPath)
+	testCases := []struct {
+		name          string
+		printable     bool
+		itemId        string
+		expectedKey   string
+		expectedValue string
+	}{
+		{
+			name:          "printable items",
+			printable:     true,
+			itemId:        "0",
+			expectedKey:   "key_0",
+			expectedValue: "value_0",
+		},
+		{
+			name:          "non printable items",
+			printable:     false,
+			itemId:        "0",
+			expectedKey:   hex.EncodeToString(convertInt64IntoBytes(0 + 1)),
+			expectedValue: hex.EncodeToString(convertInt64IntoBytes(0 + 2)),
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			db := btesting.MustCreateDBWithOption(t, &bolt.Options{PageSize: 4096})
+			srcPath := db.Path()
+
+			t.Log("Insert some sample data")
+			err := db.Update(func(tx *bolt.Tx) error {
+				b, bErr := tx.CreateBucketIfNotExists([]byte("data"))
+				if bErr != nil {
+					return bErr
+				}
 
-	meta := readMetaPage(t, srcPath)
-	leafPageId := 0
-	for i := 2; i < int(meta.Pgid()); i++ {
-		p, _, err := guts_cli.ReadPage(srcPath, uint64(i))
-		require.NoError(t, err)
-		if p.IsLeafPage() && p.Count() > 1 {
-			leafPageId = int(p.Id())
-		}
-	}
-	require.NotEqual(t, 0, leafPageId)
+				for i := 0; i < 100; i++ {
+					if tc.printable {
+						if bErr = b.Put([]byte(fmt.Sprintf("key_%d", i)), []byte(fmt.Sprintf("value_%d", i))); bErr != nil {
+							return bErr
+						}
+					} else {
+						k, v := convertInt64IntoBytes(int64(i+1)), convertInt64IntoBytes(int64(i+2))
```

Comment:
same here, why we increment the `i` ? .. not the case in the `printable` data above !